presto-native-execution/presto_cpp/main/types/tests/RowExpressionTest.cpp

```diff
@@ -278,6 +278,76 @@ TEST_F(RowExpressionTest, varchar3) {
   testConstantExpression(str, "VARCHAR", "\"102\"");
 }
 
+TEST_F(RowExpressionTest, varbinary1) {
+  // The result was generated from
+  // `select to_big_endian_32(1)`.
+  std::string str = R"##(
+        {
+            "@type": "constant",
+            "valueBlock": "DgAAAFZBUklBQkxFX1dJRFRIAQAAAAQAAAAABAAAAAAAAAE=",
+            "type": "varbinary"
+        }
+    )##";
+  testConstantExpression(str, "VARCHAR", R"("\u0000\u0000\u0000\u0001")");
+}
+
+TEST_F(RowExpressionTest, varbinary2) {
+  // The result was generated from
+  // `select cast('value' as varbinary)`.
+  std::string str = R"##(
+        {
+            "@type": "constant",
+            "valueBlock": "DgAAAFZBUklBQkxFX1dJRFRIAQAAAAUAAAAABQAAAHZhbHVl",
+            "type": "varbinary"
+        }
+    )##";
+  testConstantExpression(str, "VARCHAR", "\"value\"");
+}
+
+TEST_F(RowExpressionTest, varbinary3) {
+  // The result was generated from
+  // `select cast('SPECIAL_#@,$|%/^~?{}+-' as varbinary)`.
+  std::string str = R"##(
+        {
+            "@type": "constant",
+            "valueBlock": "DgAAAFZBUklBQkxFX1dJRFRIAQAAABYAAAAAFgAAAFNQRUNJQUxfI0AsJHwlL15+P3t9Ky0=",
+            "type": "varbinary"
+        }
+    )##";
+  testConstantExpression(str, "VARCHAR", "\"SPECIAL_#@,$|%/^~?{}+-\"");
```

Comment:
`VARCHAR`->`VARBINARY`?

Reply:
This is what Velox generated, the result is returned as string.

Reply:
I will take a closer look to see why that is the case.

Reply:
Good catch ! I had let that go because that was an interpretation of the original string literal of the cast.

The underlying reason here is that there isn't a variant for TypeKind::VARBINARY. Maybe that wasn't necessary since VARBINARY mapped to StringView as per the VariantTypeTraits and that was a shortcut. But yes, that could result in a bit weird code like this one.

We have  2 options here :
i) Introduce native variant for VARBINARY in Velox.
ii) Change the ConstantExpression construction from the JSON at https://github.com/prestodb/presto/blob/master/presto-native-execution/presto_cpp/main/types/PrestoToVeloxExpr.cpp#L414
from 
return std::make_shared<ConstantTypedExpr>(value);
to
return std::make_shared<ConstantTypedExpr>(type, value); instead.